### ISSUE TITLE
[Bug] Input settings not saving/reverting to default values

### ISSUE THREAD

**Issue by dannyglover** [Ryujinx/Ryujinx#6959]
### Description of the issue

When changing input settings for Player 2 or above, the settings simply do not save.

The changes are not reflected in-game input and also are reverted when you check the input menu again to look at the bindings.

Why is this happening? Curiously, I didn't have this issue on Fedora, but on my Windows rig it happens all the time.

### Reproduction steps

1. Open the input settings menu
2. Setup more than one controller
3. Using (for example) an xbox controller, swap the bindings for A/B, so they match the xbox controller layout. For player 2.
4. Apply/Save the settings and close the menu. Upon opening the menu again, the input values you configured are reset to default values for Player 2 onwards.

### Log file

```00:00:00.048 |I| Configuration LogValueChange: BackendThreading set to: Auto
00:00:00.049 |I| Configuration LogValueChange: ResScale set to: 1
00:00:00.056 |I| Configuration LogValueChange: ResScaleCustom set to: 1
00:00:00.056 |I| Configuration LogValueChange: MaxAnisotropy set to: -1
00:00:00.057 |I| Configuration LogValueChange: AspectRatio set to: Fixed16x9
00:00:00.148 |I| Configuration LogValueChange: GraphicsBackend set to: Vulkan
00:00:00.149 |I| Configuration LogValueChange: PreferredGpu set to: 
00:00:00.151 |I| Configuration LogValueChange: EnableDockedMode set to: True
00:00:00.155 |I| Configuration LogValueChange: EnableVsync set to: True
00:00:00.155 |I| Configuration LogValueChange: EnableShaderCache set to: True
00:00:00.155 |I| Configuration LogValueChange: EnableTextureRecompression set to: False
00:00:00.155 |I| Configuration LogValueChange: EnableMacroHLE set to: True
00:00:00.155 |I| Configuration LogValueChange: EnableColorSpacePassthrough set to: False
00:00:00.156 |I| Configuration LogValueChange: AntiAliasing set to: None
00:00:00.156 |I| Configuration LogValueChange: ScalingFilter set to: Bilinear
00:00:00.157 |I| Configuration LogValueChange: ScalingFilterLevel set to: 80
00:00:00.157 |I| Configuration LogValueChange: EnablePtc set to: True
00:00:00.157 |I| Configuration LogValueChange: EnableInternetAccess set to: False
00:00:00.157 |I| Configuration LogValueChange: EnableFsIntegrityChecks set to: True
00:00:00.157 |I| Configuration LogValueChange: FsGlobalAccessLogMode set to: 0
00:00:00.157 |I| Configuration LogValueChange: AudioBackend set to: SDL2
00:00:00.157 |I| Configuration LogValueChange: AudioVolume set to: 1
00:00:00.159 |I| Configuration LogValueChange: MemoryManagerMode set to: HostMappedUnsafe
00:00:00.159 |I| Configuration LogValueChange: ExpandRam set to: False
00:00:00.159 |I| Configuration LogValueChange: IgnoreMissingServices set to: False
00:00:00.159 |I| Configuration LogValueChange: UseHypervisor set to: True
00:00:00.160 |I| Configuration LogValueChange: MultiplayerMode set to: Disabled
00:00:00.307 |N| Application PrintSystemInfo: Ryujinx Version: 1.1.1336
00:00:00.310 |N| Application Print: Operating System: Microsoft Windows 10.0.22621 (X64)
00:00:00.311 |N| Application Print: CPU: AMD Ryzen 7 5800X3D 8-Core Processor ; 16 logical
00:00:00.312 |N| Application Print: RAM: Total 47.9 GiB ; Available 42.1 GiB
00:00:00.314 |N| Application PrintSystemInfo: Logs Enabled: Info, Warning, Error, Guest, Stub
00:00:00.315 |N| Application PrintSystemInfo: Launch Mode: UserProfile
00:00:05.846 |E| .NET TP Worker ServiceTime InitializeLocationNameCache: TimeZoneBinary system title not found! TimeZone conversions will not work, provide the system archive to fix this error. (See https://github.com/Ryujinx/Ryujinx/wiki/Ryujinx-Setup-&-Configuration-Guide#initial-setup-continued---installation-of-firmware for more information)
00:02:07.976 |I| Configuration LogValueChange: PreferredGpu set to: 0x1002_0x73A5
00:02:09.808 |E| .NET TP Worker ServiceTime InitializeLocationNameCache: TimeZoneBinary system title not found! TimeZone conversions will not work, provide the system archive to fix this error. (See https://github.com/Ryujinx/Ryujinx/wiki/Ryujinx-Setup-&-Configuration-Guide#initial-setup-continued---installation-of-firmware for more information)
00:02:17.817 |N| Application ReloadConfig: Loading configuration from: C:\Users\[redacted]\AppData\Roaming\Ryujinx\Config.json
```

### OS

Windows 11

### Ryujinx version

1.1.1336

### Game version

N/A

### CPU

Ryzen 5800X3D

### GPU

RX 6950XT

### RAM

48GB DDR4

### List of applied mods

None

**Comment by Jer-C** [comment]
Can confirm: it's working correctly for me on Linux Mint but not Windows 11.

**Comment by Jer-C** [comment]
Possibly as of 1.1.1340?

**Comment by McDaniel067** [comment]
I have version 1.1.1340 and with one xbox controller im perfect everything keeps and saves, but add a second and it gets confused and wont save. (two xbox controllers using xbox dongle with windows 11)

**Comment by Jer-C** [comment]
> I have version 1.1.1340 and with one xbox controller im perfect everything keeps and saves, but add a second and it gets confused and wont save. (two xbox controllers using xbox dongle with windows 11)

Ah... I think I only re-tested after updating with a single controller plugged in. I'm away for the US Independence Day weekend, so I won't be able to verify if that was my mistake until I'm home again.

This is probably still an issue, then.

**Comment by AdiZarfaty** [comment]
this issue still exists.
version 1.1.1346
 I have multiple controllers. its a pain.
 on relaunch of ryujinx the settings just reset to whatever/disable per player.
 when setting buttons and rumble, the configuration is not saved. rumble flag is not saved.

**Comment by antoniodesousa** [comment]
> this issue still exists. version 1.1.1346 I have multiple controllers (4 xbox controllers). its a pain. on relaunch of ryujinx the settings just reset to whatever/disable per player. when setting buttons and rumble, the configuration is not saved. rumble flag is not saved.

Same issue, save/load profiles seems broken. It has been like this for months now. It's very annoying because I have to setup my controllers again after re-launching.

**Comment by McDaniel067** [comment]
> > this issue still exists. version 1.1.1346 I have multiple controllers (4 xbox controllers). its a pain. on relaunch of ryujinx the settings just reset to whatever/disable per player. when setting buttons and rumble, the configuration is not saved. rumble flag is not saved.
> 
> 
> 
> Same issue, save/load profiles seems broken. It has been like this for months now. It's very annoying because I have to setup my controllers again after re-launching.

Agreed, every time we go to play it's an extra annoying step to do prior. I don't think the fix would be controller association as much as just the profiles sticking when saved. I wouldn't mind having to assign controllers a profile but having to do everything from scratch every time is bad.

**Comment by neil18** [comment]
Similar Issue for me, when I click OK after changing A/B/X/Y, the changes do seem to be reflected in Config.json and in-game. It only reverts to default when I open the input menu for player 2 again.

**Comment by ellisshorty** [comment]
I have the same issue can never get 2 joycons working even with manually editing the file it always reverts or doesn't save.

Anyone found a workaround for mac?
I even tried locking the config file with terminal after manually editing but then the games cant launch. 

Never been able to play 2 player, 

Sometimes I can get to map after trying 100s of times but the horizontal  mode reverts back and i need that for the game.

**Comment by gazreese** [comment]
Also seeing similar issues on Mac with 1379. 4x DualSense 4 remotes. Controllers appears as the same name with (0) (1) (2) (2) and is impossible to select one of the controllers as it's very confused about the numbering. Seems to be Player 3 that gets disables.

Sometimes the controllers appear as (0) (0) (1) (2). Just seems impossible to set controllers up for more than 2x players. 3rd player always gets disabled.

**Comment by nemajor1** [comment]
Hello, problem on aboard, when i set new settings for player2 , close and open menu sets a default setting

**Comment by McDaniel067** [comment]
> Hello, problem on aboard, when i set new settings for player2 , close and open menu sets a default setting

Yeah this happens to me too. I almost have to just set it and close the settings tab and hope the settings stuck. Very frustrating.

**Comment by 7ngel** [comment]
Same issue, devs please fix this

**Comment by Jer-C** [comment]
> Same issue, devs please fix this

Remember that this is an open source project - you can help fix issues as well!

**Comment by 7ngel** [comment]
> > Same issue, devs please fix this
> 
> Remember that this is an open source project - you can help fix issues as well!
I have a job and have no idea how to code so please let me know how I can some how create the time to work around my job and start coding to get right on that! I get the sentiment but dude but I'm literally just commenting on the ticket to let the devs know it's still an issue.

**Comment by piplup55** [comment]
> I have a job and have no idea how to code so please let me know how I can some how create the time to work around my job and start coding to get right on that! I get the sentiment but dude but I'm literally just commenting on the ticket to let the devs know it's still an issue. Also I don't see you contributing a single thing so who are you to tell me that?

any issue that has the open status we are aware of when it gets fixed it will be closed, there no reason to throw insults

**Comment by Donkcfull** [comment]
Tried to go back as far as possible. Oldest version that still supports firmware 18.0 is 1.1.359, as far as i could tell. The problem is also present in this version. Just to let you know.

**Comment by oprey123** [comment]
Still experiencing this issue.  Makes setting up multiple controllers impossible.  Each time the settings are adjusted some other controller settings disappear.
